### PR TITLE
initial commit for ubuntu-22.04 xdp

### PR DESCRIPTION
**Pull request by ami-GS**

### CODE REVIEW

File: .github/workflows/quic.yml [hunk 1]
```diff
@@ -124,8 +124,8 @@ jobs:
       os: ${{ matrix.os }}
       ref: ${{ github.event.client_payload.sha || github.event.client_payload.ref || inputs.ref || 'main' }}
 
-  build-unix:
-    name: Build Unix
+  build-ubuntu20.04:
```
Comment (nibanks) [inline .github/workflows/quic.yml:127]: Don't do it this way. Just add `ubuntu-22.04` to the `os` part of the matrix below.

File: .github/workflows/quic.yml [hunk 2]
```diff
@@ -82,7 +82,10 @@ jobs:
     name: For ${{ github.event.client_payload.guid }}
     if: ${{ github.event_name == 'repository_dispatch' }}
     needs: []
-    runs-on: ubuntu-20.04
```
Comment (nibanks) [inline .github/workflows/quic.yml:85]: You don't need to touch this. This is just for echoing input parameters.

File: .github/workflows/quic.yml [hunk 3]
```diff
@@ -239,7 +244,10 @@ jobs:
   generate-summary:
     name: Results
     needs: [run-secnetperf]
-    runs-on: 'ubuntu-20.04'
```
Comment (nibanks) [inline .github/workflows/quic.yml:242]: You don't have to run everything (below) in a matrix

File: .github/workflows/quic_matrix.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 [
     { "env": "azure", "os": "ubuntu-20.04", "arch": "x64", "tls": "openssl",  "io": "epoll" },
+    { "env": "azure", "os": "ubuntu-22.04", "arch": "x64", "tls": "openssl",  "io": "xdp" },
```
Comment (nibanks) [inline .github/workflows/quic_matrix.json:3]:
```suggestion
    { "env": "azure", "os": "ubuntu-22.04", "arch": "x64", "tls": "openssl3", "io": "xdp" },
```

Comment (nibanks) [reply .github/workflows/quic_matrix.json:3]: You should also add epoll for 22.04.

File: .github/workflows/quic.yml [hunk 5]
```diff
@@ -259,7 +261,9 @@ jobs:
     if: ${{ (github.event_name == 'repository_dispatch' && github.event.client_payload.pr == '') || inputs.commit }}
     strategy:
       fail-fast: false
-    runs-on: 'ubuntu-20.04'
```
Comment (nibanks) [inline .github/workflows/quic.yml:262]: All of these don't need to run on multiple. Keep them on 20.04. They're just processing data

File: .github/workflows/quic.yml [hunk 6]
```diff
@@ -130,16 +130,18 @@ jobs:
...
+          - os: 'ubuntu-22.04'
+            tls: 'openssl3'
     uses: microsoft/msquic/.github/workflows/build-reuse-unix.yml@main
     with:
       os: ${{ matrix.os }}
       tls: ${{ matrix.tls }}
       build: -Perf
       ref: ${{ github.event.client_payload.sha || github.event.client_payload.ref || inputs.ref || 'main' }}
 
-
```
Comment (nibanks) [inline .github/workflows/quic.yml:142]: nit: revert please

File: .github/workflows/quic_matrix.json [hunk 7]
```diff
@@ -1,5 +1,7 @@
 [
     { "env": "azure", "os": "ubuntu-20.04", "arch": "x64", "tls": "openssl",  "io": "epoll" },
+    { "env": "lab", "os": "ubuntu-22.04", "arch": "x64", "tls": "openssl3",  "io": "epoll" },
+    { "env": "lab", "os": "ubuntu-22.04", "arch": "x64", "tls": "openssl3",  "io": "xdp" },
```
Comment (nibanks) [inline .github/workflows/quic_matrix.json:4]: Please add to the end of the list with the other lab. Also, please fix formatting offset so they line up.

File: .github/workflows/quic_matrix.json [hunk 8]
```diff
@@ -1,10 +1,8 @@
 [
-    { "env": "azure", "os": "ubuntu-20.04", "arch": "x64", "tls": "openssl",  "io": "epoll" },
-    { "env": "azure", "os": "windows-2022", "arch": "x64", "tls": "schannel", "io": "iocp" },
-    { "env": "azure", "os": "windows-2022", "arch": "x64", "tls": "schannel", "io": "xdp" },
-    { "env": "azure", "os": "windows-2022", "arch": "x64", "tls": "schannel", "io": "wsk" },
```
Comment (nibanks) [inline .github/workflows/quic_matrix.json:5]: @ProjectsByJackHe are you Ok with removing these until we get the 1ES pools working?

Comment (ProjectsByJackHe) [reply .github/workflows/quic_matrix.json:5]: FYI I already added code in manage-azure-machines.ps1 to filter out the Azure VMs. This is unnecessary.

Comment (ProjectsByJackHe) [reply .github/workflows/quic_matrix.json:5]: image

Comment (ProjectsByJackHe) [reply .github/workflows/quic_matrix.json:5]: @ami-GS can you remove these changes? then I can approve

**Review by ProjectsByJackHe — APPROVED** [review]
LGTM.

**Pull request merged**